.github/workflows/sbom.yml

```diff
@@ -143,13 +149,32 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: |
           for artifact in ./sboms/*/*; do
+            # Extract the file and directory names
             base_name=$(basename "$artifact")
             dir_name=$(basename "$(dirname "$artifact")")
-            unique_name="${dir_name}-${base_name}"
-            gh release upload "$RELEASE_TAG" "$artifact" \
-              --repo "camunda/camunda" \
-              --clobber \
-              --name "$unique_name"
+            unique_name="${dir_name}-${base_name}-${RELEASE_TAG}"
+
+            # Temporarily rename the artifact with the unique name
+            mv "$artifact" "./sboms/$unique_name"
+            echo "Renamed $artifact to ./sboms/$unique_name"
+
+            # Upload the artifact
+            echo "Uploading ./sboms/$unique_name to release $RELEASE_TAG"
+            gh release upload "$RELEASE_TAG" "./sboms/$unique_name" \
+            --repo "camunda/camunda" \
+            --clobber
+
+            # Check the exit code of the upload command
+            if [ $? -eq 0 ]; then
+              echo "Successfully uploaded: $unique_name"
+            else
+              echo "Failed to upload: $unique_name"
+            exit 1
```

Comment:
Nitpick: indention seems off, since is part of the `else`-branch

Also exiting here will not restore the original artifact name (not sure if that is important to you)